ownCloud/Client/Actions/Action.swift

```diff
@@ -281,4 +321,152 @@ class Action : NSObject {
+			}
+
+			if hasLinkSharing {
+				let addGroupRow = StaticTableViewRow(rowWithAction: { (_, _) in
+					presentingController.dismiss(animated: true)
+
+					if let viewController = context.viewController {
+						let sharingViewController = PublicLinkTableViewController(core: context.core!, item: item)
+						sharingViewController.shares = shares
+						let navigationController = ThemeNavigationController(rootViewController: sharingViewController)
+						navigationController.modalPresentationStyle = .formSheet
+						viewController.present(navigationController, animated: true, completion: nil)
+					}
+				}, title: linkTitle, subtitle: nil, image: UIImage(named: "link"), alignment: .left, accessoryType: .disclosureIndicator)
+				shareRows.append(addGroupRow)
+			} else if let publicLinkRow = self.shareAsPublicLinkRow(item: item, presentingController: presentingController, context: context) {
+				shareRows.append(publicLinkRow)
+			}
+		} else if item.isShareable {
+			shareRows.append(self.shareAsGroupRow(item: item, presentingController: presentingController, context: context))
+			if let publicLinkRow = self.shareAsPublicLinkRow(item: item, presentingController: presentingController, context: context) {
+				shareRows.append(publicLinkRow)
+			}
+		}
+
+		return shareRows
+	}
+
+	class func updateSharingSection(sectionIdentifier: String, rows: [StaticTableViewRow], tableViewController: MoreStaticTableViewController) {
+		if let section = tableViewController.sectionForIdentifier(sectionIdentifier) {
+			tableViewController.removeSection(section)
+		}
+		if rows.count > 0 {
+			tableViewController.insertSection(MoreStaticTableViewSection(identifier: "share-section", rows: rows), at: 0, animated: false)
+		}
+	}
+
+	class func shareAsGroupRow(item : OCItem, presentingController: UIViewController, context: ActionContext) -> StaticTableViewRow {
+		var title = "Share this file".localized
+		if item.type == .collection {
+			title = "Share this folder".localized
+		}
+
+		let addGroupRow = StaticTableViewRow(buttonWithAction: { (_, _) in
+			presentingController.dismiss(animated: true)
+
+			if let viewController = context.viewController {
+				let sharingViewController = GroupSharingTableViewController(core: context.core!, item: item)
+				let navigationController = ThemeNavigationController(rootViewController: sharingViewController)
+				navigationController.modalPresentationStyle = .formSheet
+
+				viewController.present(navigationController, animated: true, completion: nil)
+			}
+		}, title: title, style: .plain, image: UIImage(named: "group"), alignment: .left, identifier: "share-add-group")
+
+		return addGroupRow
+	}
+
+	class func shareAsPublicLinkRow(item : OCItem, presentingController: UIViewController, context: ActionContext) -> StaticTableViewRow? {
+		if context.core!.connection.capabilities?.publicSharingEnabled == true, item.sharedByPublicLink() == false, item.isShareable {
```

Comment:
Comparisons like `== true` and `== false` are most useful when you deal with an Optional (`Bool?`) value. When dealing with a plain `Bool`, `!value` or `value` is often preferable:
```suggestion
		if context.core!.connection.capabilities?.publicSharingEnabled == true, !item.sharedByPublicLink(), item.isShareable {
```